integration-cli/docker_cli_build_test.go

```diff
@@ -5660,3 +5660,434 @@ func (s *DockerSuite) TestBuildNullStringInAddCopyVolume(c *check.C) {
 	_, err = buildImageFromContext(name, ctx, true)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	args := []string{
+		"--build-arg", fmt.Sprintf("%s=%s", envKey, envVal),
+	}
+	dockerfile := fmt.Sprintf(`FROM busybox
+		ARG %s=
```

Comment:
OK, I was under the impression we would only allow the builtin ones to have no default values but force Dockerfile authors to provide a default.

The only concern I have is if for some reason we would ever want to make some build args required in the future, the `ARG foo` syntax could be handy. Although the current design is precisely against build args being required.

Genuine question: are we sure we will never need required build args and that allowing to have no default values for non-builtin build args is so important for a first iteration? /cc @icecrime